src/components/Banner/Banner.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import classNames from 'classnames';
+import _ from 'lodash';
+import { bindClassNames } from '../../util/style-helpers';
+
+const boundClassNames = bindClassNames('Banner');
+
+const {
+	bool,
+	string,
+	oneOfType,
+	oneOf,
+	node,
+	arrayOf,
+	func,
+	element
+} = React.PropTypes;
+
+const defaultIcons = {
+	'success': '<SuccessIcon/>',
+	'danger': '<DangerIcon/>',
+	'info': '<InfoIcon/>',
+	'warning': '<WarningIcon/>',
+	'primary': null,
+	'default': null
+};
+
+/**
+ *
+ * {"categories": ["controls", "Banners"]}
+ *
+ * A basic Banner. Any props that are not explicitly called out below will be
+ * passed through to the native `Banner` component.
+ */
+const Banner = React.createClass({
+	propTypes: {
+		/**
+		 * pass in a bool to display predefined icon based on `kind`.
+		 */
+		hasIcon: bool,
+		/**
+		 * pass in a icon component for custom icons within banner.
+		 */
+		icon: element,
+		/**
+		 * set this to `true` if you want to have a `x` close icon.
+		 */
+		isCloseable: bool,
+		/**
+		 * set this to `true` if you want to remove rounded corners.
+		 */
+		isFullWidth: bool,
+		/**
+		 * class names that are appended to the defaults
+		 */
+		className: string,
+		/**
+		 * any valid React children
+		 */
+		children: oneOfType([
+			node,
+			arrayOf(node)
+		]),
+		/**
+		 * style variations of the Banner
+		 */
+		kind: oneOf([
+			'primary',
+			'success',
+			'warning',
+			'danger',
+			'info'
+		]),
+		/**
+		 * size variations of the Banner
+		 */
+		size: oneOf([
+			'small'
+		]),
+		/**
+		 * called when the user clicks the Banner
+		 */
+		onClick: func,
+	},
+
+	getDefaultProps() {
+		return {
+			hasIcon: false,
+			icon: null,
+			isCloseable: true,
+			isFullWidth: false,
+			kind: 'default',
+			onClick: _.noop
+		};
+	},
+
+	handleClick() {
+		this.props.onClick();
```

Comment:
Since we aren't doing anything special for `onClick` we should just pass `this.props.onClick` directly without a handler.